lib/player.js

```diff
@@ -2034,6 +2034,16 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
         });
       },
       getBandwidthEstimate: () => this.abrManager_.getBandwidthEstimate(),
+      onMetadata: (type, startTime, endTime, values) => {
+        let metadataType = type;
+        if (type == 'com.apple.hls.interstitial') {
+          metadataType = 'com.apple.quicktime.HLS';
+        }
+        for (const payload of values) {
+          this.dispatchMetadataEvent_(
```

Comment:
As-is, this might play weird with preload. If you start to preload an asset with `EXT-X-DATERANGE`, the player itself will start to emit `metadata` events, I believe. That could be misleading for developers if you are preloading an asset while another one is playing.
If we wrap this loop with `preloadManager.addQueuedOperation(false, ...)`, it will delay the `metadata` events until the preloader has begun to load.

Reply:
Done!